packages/tldraw/src/lib/shapes/geo/components/GeoShapeBody.tsx

```diff
@@ -209,14 +216,19 @@ export function GeoShapeBody({ shape }: { shape: TLGeoShape }) {
 				const d = getDrawHeartPath(w, h, strokeWidth, shape.id)
 				return (
 					<>
-						<ShapeFill theme={theme} d={d} color={color} fill={fill} scale={scale} />
+						<ShapeFill theme={theme} d={d} color={color} fill={fill} scale={scaleToUse} />
 						<path d={d} stroke={theme[color].solid} strokeWidth={strokeWidth} fill="none" />
 					</>
 				)
 			}
 		}
 		default: {
-			const geometry = editor.getShapeGeometry(shape)
+			let geometry = editor.getShapeGeometry(shape)
+			if (!shouldScale) {
+				// We don't want the cached geometry
+				geometry = editor.getShapeUtil(shape).getGeometry(shape)
```

Comment:
I don't love this, but couldn't come up with a better way

Reply:
Why would the cached geometry be wrong here?